src/computer_session.cpp

```diff
@@ -536,9 +536,10 @@ void computer_session::action_cascade()
 
 void computer_session::action_research()
 {
+    map &here = get_map();
     // TODO: seed should probably be a member of the computer, or better: of the computer action.
     // It is here to ensure one computer reporting the same text on each invocation.
-    const int seed = g->get_levx() + g->get_levy() + g->get_levz() + comp.alerts;
+    const int seed = here.get_abs_sub().x + here.get_abs_sub().y + here.get_abs_sub().z + comp.alerts;
```

Comment:
Position could be a variable and have one call to `get_abs_sub()`.

Reply:
Better yet: use e.g. `std::hash<tripoint>{}( here.get_abs_sub() )` rather than rolling a poor-quality hash function by hand.